src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -4726,6 +4750,30 @@ int wm_vuldet_set_agents_info(agent_software **agents_software, update_node **up
         } else if (strcasestr(os_name, vu_feed_ext[FEED_WIN])) {
             agent_dist_ver = wm_vuldet_decode_win_os(os_name);
             agent_dist = FEED_WIN;
+        } else if (strcasestr(os_platform, vu_feed_ext[FEED_MAC])) {
+            char *version = NULL;
+            w_strdup(agent_version, version);
+
+            if (agent_version) {
+                strtok(agent_version, "v");
+                agent_major=strtok(NULL, ".");
+                agent_minor=strtok(NULL, ".");
+                if (agent_major == NULL) {
+                    mdebug1("Cannot parse agent_major from version agent");
+                }
+                if (agent_minor == NULL) {
+                    mdebug1("Cannot parse agent_minor from version agent");
+                }
+            }
+
+            if (atoi(agent_major) > 4 || (atoi(agent_major) == 4 && atoi(agent_minor) > 0)) {
```

Comment:
Be careful here when `agent_major` or `agent_minor` is NULL. For those cases, these operations should be skipped.

In addition, the second condition would not work for versions higher than 4.X, for example, 5.0.